### PR TITLE
Map JDBC's Types.OTHER to VARCHAR in base-jdbc connector

### PR DESCRIPTION
**Pull request by albertocsm**
instead of discarding unknown column types, try to convert it to an UTF_8 String.
it will produce a readable result if the value being returned is a byte[](handled for *free* by io.airlift.slice)

Refs #4082

### CODE REVIEW

**Comment by electrum** [comment]
Can you add a test for this?

**Comment by albertocsm** [comment]
would love to. not sure where...

**Comment by electrum** [comment]
I don't think this change will affect MySQL, since from what I can tell, all the types they support (at least in versions before 5.7) are covered by standard JDBC types.

We can add a PostgreSQL specific test in `TestPostgreSqlDistributedQueries` that creates a table with a type that their driver handles as `OTHER` (maybe `uuid` or `inet`?), inserts some data and queries it.

**Comment by electrum** [comment]
Note that by default, the command line build does not run those tests, since they take a long time. They only run if the Maven `ci` profile is activated. You can run the tests for the JDBC connectors like this:
```
mvn clean install -P ci -pl presto-base-jdbc,presto-mysql,presto-postgresql
```
Or you can run individually for each module. Just make sure to do an `install` for `presto-base-jdbc` before running `presto-postgresql` in order for it to pick up the change in the base module (if you do all at once, Maven handles the dependency order correctly).

**Comment by albertocsm** [comment]
@electrum plz take a look

**Comment by albertocsm** [comment]
@electrum plz take a look

**Comment by albertocsm** [comment]
Can i do something more to move this forward?

**Comment by FlavioF** [comment]
Any new about this PR? It will be merged? I am looking for this too.

**Comment by jomach** [comment]
I have the same problem but on select. 
jorges-mbp:Downloads jorge$ ./presto_cli --server 192.168.99.100:8080 --catalog postgresql --schema public

show tables work !

presto:public> select * from public.datasets;
Query 20160310_130418_00088_9jyhv failed: Table has no supported column types: public.datasets

**Comment by albertocsm** [comment]
rebased to master's current revision.
can some one plz review this?

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java [hunk 1]
```diff
@@ -78,6 +79,24 @@ public void testViews()
         execute("DROP VIEW IF EXISTS tpch.test_view");
     }
 
+    @Test
+    public void testSelectNativeDataTypeColumn()
+            throws Exception
+    {
```
Comment (maciejgrzybek) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java]: Please add preconditions check to ensure `test_other_data_type` table does not exist before you attempt creating it.

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java [hunk 2]
```diff
@@ -78,6 +79,24 @@ public void testViews()
...
+        execute("CREATE TABLE tpch.test_other_data_type (_foo UUID)");
```
Comment (maciejgrzybek) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java]: To be more standardized, please use `assertUpdate`, similarly to `TestMySqlIntegrationSmokeTest`.

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java [hunk 3]
```diff
@@ -78,6 +79,24 @@ public void testViews()
...
+        assertTrue(queryRunner.tableExists(getSession(), "test_other_data_type"));
+
+        execute("INSERT INTO tpch.test_other_data_type VALUES ('00000000-0000-0000-0000-000000000000')");
```
Comment (maciejgrzybek) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java]: Please use `assertUpdate` instead.

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java [hunk 4]
```diff
@@ -78,6 +79,24 @@ public void testViews()
...
+        assertEquals(
```
Comment (maciejgrzybek) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java]: Maybe consider using `assertQuery` to ensure not only row count but also type match and returned result?

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java [hunk 5]
```diff
@@ -78,6 +79,24 @@ public void testViews()
...
+            queryRunner
+                .execute(getSession(), "SELECT _foo from test_other_data_type")
+                .getRowCount(),
+            1);
+
+        execute("DROP TABLE IF EXISTS tpch.test_other_data_type");
```
Comment (maciejgrzybek) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java]: ditto `assertUpdate`

**Comment by albertocsm** [comment]
will do. thks for the input.

**Comment by cberner** [comment]
@albertocsm are you still working on this?

**Comment by albertocsm** [comment]
Haven't had time recently
I'll try to give a look until the end of the week if that's ok for you

File: presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java [hunk 6]
```diff
@@ -78,6 +79,24 @@ public void testViews()
         execute("DROP VIEW IF EXISTS tpch.test_view");
     }
 
+    @Test
+    public void testSelectNativeDataTypeColumn()
```
Comment (findepi) [inline presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlDistributedQueries.java]: Please add your tests to `TestPostgreSqlTypeMapping` and `TestMySqlTypeMapping`

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java [hunk 7]
```diff
@@ -427,6 +427,7 @@ protected Type toPrestoType(int jdbcType)
             case Types.NVARCHAR:
             case Types.LONGVARCHAR:
             case Types.LONGNVARCHAR:
+            case Types.OTHER:
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java]: If we can (optimistically) map `Types.OTHER` to `varchar`, then maybe we can map _others_ (`default`) to `varchar`? (I'm not suggesting, I'm just thinking aloud and asking for your thoughts)

Either way, what do we do if this turns out to be bad guess, and the `ResultSet.getString` doesn't work for one of the now-seemingly-supported data types?

**Comment by findepi** [comment]
@albertocsm it would be better not to have catch-all mapping for other. Adding support for Postgres types like `UUID` or `money` is fairly easy, but each mapping should be covered by an explicit test.
I would rather see the mappings in postgresql connector, even using postgresql-drive-specific APIs if necessary.

(I'm closing this now due to little activity for a long time. Please reopen or new PR whenever you wish to continue with that.)

**Pull request closed without merging**